common/nodejs-utils/src/download_file.rs

```diff
@@ -0,0 +1,106 @@
+use reqwest::IntoUrl;
+use reqwest_middleware::ClientBuilder;
+use reqwest_retry::policies::ExponentialBackoff;
+use reqwest_retry::RetryTransientMiddleware;
+use std::fs::File;
+use std::io;
+use std::path::{Path, PathBuf};
+use std::time::Duration;
+
+const DEFAULT_CONNECT_TIMEOUT: Duration = Duration::from_secs(5);
```

Comment:
In https://github.com/heroku/heroku-buildpack-ruby/pull/175 a customer asked to increase the default from 3 to 30 seconds. I ended up making it configurable. I think someone running `pack build` in Australia on a tetherd 3g mobile should be slow, but should be possible. 

I think we need to expose a way to allow customers to increase this value if they need, either now or add an issue to our backlog. My suggestion would be something in `project.toml` like `com.heroku.connect.timeout = 30` (spitballing).

Reply:
It could be configurable. I don't think it should block this PR though. We should discuss this more broadly with the team, especially if it's going to be a generalized configuration like `com.heroku.connect.timeout`.